### PR TITLE
Add log-in module UI & functionality

### PR DESCRIPTION
**Pull request by Naoe-Adrian**
I add validator on login page

### CODE REVIEW

File: SpaceShare.UI/src/app/landing/landing.component.html [hunk 1]
```diff
@@ -1,16 +1,29 @@
...
   </div>
 </div>
 
+<!-- Modal Trigger -->
+<input type="checkbox" id="my_modal_6" class="modal-toggle" />
+<div class="modal">
+  <div class="modal-box">
+    <app-login></app-login>
+    <div class="modal-action">
+      <label for="my_modal_6"class="btn btn-sm btn-circle btn-ghost absolute right-2 top-2">X</label>
```
Comment (Galope-Jhean) [inline SpaceShare.UI/src/app/landing/landing.component.html:21]:
```suggestion
      <label for="my_modal_6"class="btn btn-sm btn-circle btn-ghost absolute right-2 top-2">&#x2715;</label>
```

File: SpaceShare.UI/src/app/landing/login/login.component.html [hunk 2]
```diff
@@ -1 +1,31 @@
...
+  <form class="card-body" [formGroup]="loginForm" (ngSubmit)="onSubmit()">
+    <div class="form-control">
+        <label class="input input-bordered flex items-center gap-2 rounded-md mb-3" [class.input-error]="emailControl.invalid && emailControl.touched">
+            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16" fill="currentColor" class="w-4 h-4 opacity-70">
+                <path d="M2.5 3A1.5 1.5 0 0 0 1 4.5v.793c.026.009.051.02.076.032L7.674 8.51c.206.1.446.1.652 0l6.598-3.185A.755.755 0 0 1 15 5.293V4.5A1.5 1.5 0 0 0 13.5 3h-11Z" />
+                <path d="M15 6.954 8.978 9.86a2.25 2.25 0 0 1-1.956 0L1 6.954V11.5A1.5 1.5 0 0 0 2.5 13h11a1.5 1.5 0 0 0 1.5-1.5V6.954Z" />
+            </svg>
+            <input type="email" formControlName="email" placeholder="Email" (focus)="clearError('email')"/>
+        </label>
+      <div *ngIf="emailControl.invalid && emailControl.touched" class="error-message text-red-500">Invalid email format</div>
```
Comment (Galope-Jhean) [inline SpaceShare.UI/src/app/landing/login/login.component.html:13]:
```suggestion
      @if(emailControl.invalid && emailControl.touched) {
          <div class="error-message text-red-500">Invalid email format</div>
      }
```

File: SpaceShare.UI/src/app/landing/login/login.component.html [hunk 3]
```diff
@@ -1 +1,31 @@
...
+    </div>
+    <div class="form-control">
+      <label class="input input-bordered flex items-center gap-2 rounded-md mb-4 " [class.input-error]="passwordControl.invalid && passwordControl.touched">
+        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16" fill="currentColor" class="w-4 h-4 opacity-70">
+          <path fill-rule="evenodd" d="M14 6a4 4 0 0 1-4.899 3.899l-1.955 1.955a.5.5 0 0 1-.353.146H5v1.5a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1-.5-.5v-2.293a.5.5 0 0 1 .146-.353l3.955-3.955A4 4 0 1 1 14 6Zm-4-2a.75.75 0 0 0 0 1.5.5.5 0 0 1 .5.5.75.75 0 0 0 1.5 0 2 2 0 0 0-2-2Z" clip-rule="evenodd" />
+        </svg>
+        <input type="password" formControlName="password" class="grow" placeholder="Password" (focus)="clearError('password')"/>
+      </label>
+      <div *ngIf="passwordControl.invalid && passwordControl.touched" class="error-message text-red-500">Password is required</div>
```
Comment (Galope-Jhean) [inline SpaceShare.UI/src/app/landing/login/login.component.html:22]:
```suggestion
      @if(passwordControl.invalid && passwordControl.touched) {
        <div class="error-message text-red-500">Password is required</div>
      }
```

**Review by Galope-Jhean — CHANGES_REQUESTED** [review]

File: SpaceShare.UI/src/app/landing/login/login.component.html (hunk 2 above)
Comment (Naoe-Adrian) [reply SpaceShare.UI/src/app/landing/login/login.component.html:13]: Thank you I will implement this

**Review by Galope-Jhean — APPROVED** [review]

**Pull request merged**